blockchain/blob_storage.go

```diff
@@ -0,0 +1,259 @@
+// Copyright 2025 The Kaia Authors
+// This file is part of the Kaia library.
+//
+// The Kaia library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Kaia library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Kaia library. If not, see <http://www.gnu.org/licenses/>.
+
+package blockchain
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/kaiachain/kaia/blockchain/types"
+	"github.com/kaiachain/kaia/node"
+	"github.com/kaiachain/kaia/rlp"
+)
+
```

Comment:
Let's keep your description @tnasu 

```
// Blob files are organized in subdirectories to improve file system performance:
// {baseDir}/
//   {blockNumber/1000}/                  bucket
//     {blockNumber}_{txIndex}.bin   blob file per transaction
// The existing LevelDB is inefficient for frequent pruning.
```

Reply:
The variable declarations BLOCKS_PER_BUCKET and BLOB_SIDECARS_RETENTION should be enough.